### PR TITLE
feat: export dashboard csv

### PR DESCRIPTION
**Pull request by rephus**
Closes: #9482

### Description:

[Screencast from 03-04-24 11:39:59.webm](https://github.com/lightdash/lightdash/assets/1983672/6a31d901-1305-41b0-be2c-ceba0d513fbc)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/common/modal/DashboardExportModal.tsx [hunk 1]
```diff
@@ -1,28 +1,70 @@
...
 
-export const DashboardExportModal: FC<Props & ModalProps> = ({
-    opened,
+type CsvExportProps = {
+    dashboard: Dashboard;
+};
+
+const CsvExport: FC<CsvExportProps & ModalProps> = ({ dashboard, onClose }) => {
+    const exportCsvDashboardMutation = useExportCsvDashboard();
+    const dashboardFilters: SchedulerFilterRule[] | undefined = [];
```
Comment (rephus) [inline packages/frontend/src/components/common/modal/DashboardExportModal.tsx:35]: I still need to get the `current` dashboard filters to send it to the backend, in the image we use the location, but we can't use a plain string here, any ideas @joaoviana  ?

File: packages/backend/src/services/CsvService/CsvService.ts [hunk 2]
```diff
@@ -816,4 +816,53 @@ export class CsvService {
             throw e;
         }
     }
+
+    async exportCsvDashboard(
+        user: SessionUser,
+        dashboardUuid: string,
+        queryFilters: SchedulerFilterRule[] | undefined,
```
Comment (rephus) [inline packages/backend/src/services/CsvService/CsvService.ts:823]: I copied this from the `getCsvsForDashboard` ,b ut we can change it if needed

**Comment by rephus** [comment]
This method runs in the API , not in scheduler, this should be ok, as long as we only get results from the `table` and not `all`, if we change this option we'll need to move it to the scheduler to avoid blocking the API

**Review by TuringLovesDeathMetal — APPROVED** [review]
Nice! looks great :) 

Just double checking: have we updated the analytics event for dashbaord downloads to include `csv` as a type? Also, can we update the docs please 🙏

**Review by joaoviana — CHANGES_REQUESTED** [review]
Looking great! 
Few requested changes 👍

File: packages/frontend/src/components/common/modal/DashboardExportModal.tsx [hunk 3]
```diff
@@ -84,43 +172,39 @@ export const DashboardExportModal: FC<Props & ModalProps> = ({
...
+                    w="50%"
+                    mb="xs"
+                    defaultValue="image"
+                    onChange={setExportType}
+                />
+                {exportType === 'csv' && (
+                    <CsvExport
+                        dashboard={dashboard}
+                        onClose={onClose}
+                        opened={true}
```
[the remark this answers is not in the document]
Comment (rephus) [reply packages/frontend/src/components/common/modal/DashboardExportModal.tsx:196]: mmm, we do if we take `& ModalProps` in the component

**Comment by rephus** [comment]
@joaoviana thanks, that worked, 
Screenshot from 2024-04-04 08-56-37

Before:

Screenshot from 2024-04-04 08-56-08

After:

Screenshot from 2024-04-04 09-04-08

**Review by joaoviana — APPROVED** [review]
🎉

**Pull request merged**